plugin/keyring_vault/keyring_vault-test/keyring_vault-test.cc

```diff
@@ -15,7 +17,29 @@ static int number_of_keys_generated= 0;
 static int max_generated_key_length= 0;
 static int number_of_keys_to_generate= 0;
 
-static mysql_mutex_t  LOCK_verbose;
+static mysql_mutex_t LOCK_verbose;
+static mysql_mutex_t LOCK_keys_in_keyring; 
+struct Key_in_keyring
+{
+  Key_in_keyring(const char *key_id, const char *user_id)
+    : key_id(key_id)
+    , user_id(user_id)
+    , key_signature(this->key_id + this->user_id)  
+  {}
+
+  Key_in_keyring()
+  {}
+
+  bool operator()(const Key_in_keyring &key1, const Key_in_keyring &key2)
+  {
+    return key1.key_signature < key2.key_signature;
+  }
+
+  std::string key_id;
+  std::string user_id;
+  std::string key_signature;
+};
+std::set<Key_in_keyring, Key_in_keyring> keys_in_keyring;
```

Comment:
Using the same class for comparator is not a very good idea. Comparators are copied and default-constructed in a number of places inside set/map members and STL algorithms using them.
Please, define separate class `Key_in_keyring_comparator`